### PR TITLE
Add E0269 error explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Manishearth

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -989,6 +989,40 @@ enum Method { GET, POST }
 ```
 "##,
 
+E0229: r##"
+An associated type binding was done outside of `<..>` and `where` clauses.
```
Comment (Manishearth) [inline src/librustc/diagnostics.rs]: `<...>` is thetype parameter declaration

File: src/librustc/diagnostics.rs [hunk 2]
```diff
@@ -989,6 +989,40 @@ enum Method { GET, POST }
...
+impl Foo for isize {
+    type A = usize;
+    fn boo(&self) -> usize { 42 }
+}
+
+fn baz<I>(x: &<I as Foo<A=Bar>>::A) {}
+// error: associated type bindings are not allowed here
+```
+
+To solve this error, please move the type bindings in `<..>` clause:
```
Comment (Manishearth) [inline src/librustc/diagnostics.rs]: same change here

File: src/librustc/diagnostics.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/librustc/diagnostics.rs]: I found it super nicely said like this haha.

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**